exonum/src/node/consensus.rs

```diff
@@ -708,6 +717,9 @@ impl NodeHandler {
 
     /// Handles request timeout by sending the corresponding request message to a peer.
     pub fn handle_request_timeout(&mut self, data: &RequestData, peer: Option<PublicKey>) {
+        if !self.is_enabled {
+            return;
+        }
```

Comment:
This is most likely over engineered, but a fun way to avoid this repetitive `is_enabled` checks would be to have, instead of `is_enabled`, the field like

```
consensus_handler: Option<ConsensusHander>
```

and make these `handle_xxx` methods the methods of `ConsensusHandler`, so that you simply can't call them if consensus is not enabled.

Reply:
Personally I think it'll be over-engineering.